hazelcast-client-new/src/test/java/com/hazelcast/client/quorum/ClientTransactionalMapQuorumTest.java

```diff
@@ -0,0 +1,379 @@
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+    @Test
+    public void testTxContainsKeyThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.containsKey("foo");
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+
+    @Test
+    public void testTxIsEmptyThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.isEmpty();
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+    @Test
+    public void testTxKeySetThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.keySet();
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+    @Test
+    public void testTxKeySetWithPredicateThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.keySet(TruePredicate.INSTANCE);
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+    @Test
+    public void testTxValuesThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.values();
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
```

Comment:
Why are you rolling back transaction? They are not so many tests it leads to a memory leak and 50% of the test code is related to exception handling.

This is a lot simpler:
```
@Test(expected = TransactionException.class)
public void testTxValuesThrowsExceptionWhenQuorumSizeNotMet() {
     TransactionContext transactionContext = c4.newTransactionContext(options);
     transactionContext.beginTransaction();
     TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
     map.values();
     transactionContext.commitTransaction();
}
```

And if you move the first line to the setup method, you test becomes:
```
@Test(expected = TransactionException.class)
public void testTxValuesThrowsExceptionWhenQuorumSizeNotMet() {
     transactionContext.beginTransaction();
     TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
     map.values();
     transactionContext.commitTransaction();
}
```